addons/binding/org.openhab.binding.ihc/src/main/java/org/openhab/binding/ihc/internal/handler/IhcHandler.java

```diff
@@ -0,0 +1,920 @@
+                } catch (IhcExecption e) {
+                    logger.error("Can't update channel '{}' value, cause ", channelUID, e.getMessage());
+                }
+            });
+        } else {
+            logger.warn("Channel {} update failed.", channelUID);
+        }
+    }
+
+    /**
+     * Update resource value to IHC controller.
+     */
+    private boolean updateResource(WSResourceValue value) throws IhcExecption {
+        boolean result = false;
+        try {
+            result = ihc.resourceUpdate(value);
+        } catch (IhcExecption e) {
+            logger.warn("Value could not be updated - retrying one time: {}", e.getMessage());
+            result = ihc.resourceUpdate(value);
+        }
+        return result;
+    }
+
+    @Override
+    public void channelLinked(ChannelUID channelUID) {
+        logger.debug("channelLinked: {}", channelUID);
+
+        switch (channelUID.getId()) {
+            case CHANNEL_CONTROLLER_STATE:
+                updateControllerStateChannel();
+                break;
+
+            case CHANNEL_CONTROLLER_UPTIME:
+                updateControllerInformationChannels();
+                break;
+
+            case CHANNEL_CONTROLLER_TIME:
+                updateControllerTimeChannels();
+                break;
+
+            default:
+                try {
+                    ChannelParams params = new ChannelParams(thing.getChannel(channelUID.getId()));
+                    if (params.getResourceId() != null) {
+                        if (!linkedResourceIds.contains(params.getResourceId())) {
+                            logger.debug("New channel '{}' found, resource id '{}'", channelUID.getAsString(),
+                                    params.getResourceId());
+                            linkedResourceIds.add(params.getResourceId());
+                            updateNotificationsRequestReminder();
+                        }
+                    }
+                } catch (IllegalArgumentException e) {
+                    logger.warn("Can't find resource id, reason:  {}", e.getMessage());
+                }
+        }
+    }
+
+    @Override
+    public void channelUnlinked(ChannelUID channelUID) {
+        logger.debug("channelUnlinked: {}", channelUID);
+
+        switch (channelUID.getId()) {
+            case CHANNEL_CONTROLLER_STATE:
+            case CHANNEL_CONTROLLER_UPTIME:
+            case CHANNEL_CONTROLLER_TIME:
+                break;
+
+            default:
+                try {
+                    ChannelParams params = new ChannelParams(thing.getChannel(channelUID.getId()));
+                    if (params.getResourceId() != null) {
+                        linkedResourceIds.removeIf(c -> c.equals(params.getResourceId()));
+                        updateNotificationsRequestReminder();
+                    }
+                } catch (IllegalArgumentException e) {
+                    logger.warn("Can't find resource id, reason:  {}", e.getMessage());
```

Comment:
You ignore stack-traces everywhere in this file, please review carefully :-)

Reply:
Done